website/source/docs/platform/k8s/mesh-gateways.html.md

```diff
@@ -0,0 +1,360 @@
+through to the correct services running in Kubernetes.
+
+Since gateways need to be accessed from other datacenters they will need to be exposed. The
+recommended way to do this is via a load balancer.
+
+### Prequisites
+The following conditions must be satisfied for gateways to work between Consul datacenters
+in Kubernetes:
+
+1. Both clusters are running Consul **v1.6.0-beta1** since gateways are a beta feature.
+    This can be set via the helm chart:
+    
+    ```yaml
+    global:
+      image: consul:1.6.0-beta1
+    ```
+1. Each cluster has a unique datacenter name, ex. `dc1` and `dc2`.
+    This can be set via the helm chart:
+    
+    ```yaml
+    global:
+      datacenter: dc2 # use dc1 in your other datacenter.
+    ``` 
+    
+1. The Consul datacenters are joined. This will require that each Consul server agent
+   is accessible from each datacenter's server agents and that the servers are
+   configured to join with the other datacenter. See our [Multiple Datacenters Guide](https://learn.hashicorp.com/consul/security-networking/datacenters).
+1. The `primary_datacenter` setting is set to a single datacenter. This is required
+   for Connect requests to work across datacenters. This can be set via the helm chart:
+   
+    ```yaml
+    client:
+      extraConfig: |
+        {
+          "primary_datacenter": "dc1"
+        }
+    server:
+      extraConfig: |
+        {
+          "primary_datacenter": "dc1"
+        }
+    ```
+1. If using Kubernetes, Connect injection must be enabled. This can be set via the helm chart:
+
+    ```yaml
+    connectInject:
+      enabled: true
+    ```
+1. gRPC must be enabled. This can be set via the helm chart:
+
+    ```yaml
+    client:
+      grpc: true
+    ```
+1. If setting a global default [gateway mode](/docs/connect/mesh_gateway.html#modes-of-operation),
+   central config must be enabled for all agents so that they actually use the default mode.
+   This can be set via the helm chart:
+
+    ```yaml
+    connectInject:
+      centralConfig:
+       enabled: true
+    ```
+
+## Guide to Using Mesh Gateways To Connect Two Kubernetes Clusters
+This guide shows how to configure the consul-helm chart to set up gateways
+that route Connect traffic between two Kubernetes clusters each running their
+own Consul datacenter.
+
+### Prequisites
+The prerequisites [specified above](/docs/platform/k8s/mesh-gateways.html#prequisites)
+must be met.
+
+### Step 1 - Get v0.9.0-beta1
+Since Gateways are still in Beta, you'll need a specific consul-helm version.
+
+```sh
+# Clone the chart repo
+git clone https://github.com/hashicorp/consul-helm.git
+cd consul-helm
+
+# Checkout the v0.9.0-beta1 release.
+git checkout v0.9.0-beta1
+
+#  You should see:
+#  Note: checking out 'v0.9.0-beta1'.
+#
+#  You are in 'detached HEAD' state. You can look around, make experimental
+#  ...
+```
+
+### Step 2 - Configure Your First DC
```

Comment:
It will only make changes to what's different. So if I haven't changed any of the, say, client config, then they won't be affected. Depending on what is changed, user intervention might be required. For example, changing the extraConfig section requires manually restarting or reloading each component.